guides/typed-json/json-data-bindings/README.md

```diff
@@ -1,58 +1,131 @@
+To bind a Typed JSON object to a database object, the `Data` property is used. 
 
-In the following example the binding will succeed since the property `Name` exists in the database object:
+Consider the following JSON file:
 
-<div class="code-name">PersonJson.json</div>
+<div class="code-name">PersonPage.json</div>
 
-```javascript
+```json
 {
-    "Name": "John"
+   "FirstName": "",
+   "LastName": "",
 }
 ```
 
-<div class="code-name">Program.cs</div>
+To bind the Typed JSON object `PersonPage` defined above to a database class `Person`, the following code can be used:
+
+<div class="code-name">Program.cs</div> 
 
 ```cs
-[Database]
-public class Person
+using Starcounter;
+
+namespace MyApp
 {
-	public string Name;
+    [Database]
+    public class Person
+    {
+        public string FirstName;
+        public string LastName;
+        public string FullName => FirstName + " " + LastName;
+    }
+
+    class Program
+    {
+        static void Main()
+        {
+            Db.Transact(() => // Add a new Person instance to the database
+            {
+                new Person()
+                {
+                    FirstName = "Steve",
+                    LastName = "Smith"
+                };
+            });
+
+            Handle.GET("/GetPerson", () =>
+            {
+                var person = Db.SQL<Person>("SELECT P FROM Person P").First; // Retrieve a database object from the database
+
+                var json = new PersonPage() 
+                {
+                    Data = person // Bind the database object to the Typed JSON object
+                };
+
+                return json;
+            });
+        }
+    }
 }
+```
+
+The `PersonPage` object will now look like this: `{"FirstName":"Steve","LastName":"Smith","FullName":"Steve Smith"}`.
 
-public class Program
+Starcounter recognizes that the properties in `PersonPage` and `Person` object have the same name and populates the values in the Typed JSON accordingly. This is the default way that bindings are created.
+
+### Binding to Code-Behind Properties
+
+In addition to binding to database objects, Typed JSON properties can also be bound to code-behind properties.
+
+To accomplish what was demonstrated in the [previous example](#binding-to-database-objects) by using code-behind properties instead of database properties, the following code can be used:
+
+<div class="code-name">PersonPage.json</div>
+
+```json
 {
-  public static void Main()
-  {
-    Handle.GET("/person/any", () =>
+   "FirstName": "",
+   "LastName": "",
+   "FullName": ""
+}
+```
+
+<div class="code-name">PersonPage.json.cs</div>
+
+```cs
+using Starcounter;
+
+namespace MyApp
+{
+    partial class PersonPage : Json
     {
-      return new PersonJson()
-      {
-        Data = Db.SQL<Person>("SELECT p FROM Person p").First;
-      });
+        public string FirstName = "Steven";
+        public string LastName = "Smith";
```

Comment:
I'm not sure about this example (meaning declaring `FirstName` and `LastName` in the code-behind). Storing values directly in the code-behind is possible, true, but I think it's a pattern we shouldn't encourage. 

Sometimes you might have to do it, but it will have implications later when we move the actual storage of the view-models outside of the current codehost/process.

Just using the `FullName` property, which simply reads `FirstName` and `LastName` should be enough to visualize the pattern.